Rebease 1.19.14 3

damemi opened the pull request:
Replacing https://github.com/openshift/kubernetes/pull/960 (hit some merge conflicts with in-flight PRs and just preferred to start over)

Commits list: https://docs.google.com/spreadsheets/d/1iLGl84N0LJv_zqoe8Vkc9TCfCiRZP8VnfSP7sbelEzE/edit#gid=1947950828&fvid=683661082

Bumps to other repos:
* https://github.com/openshift/library-go/pull/1214
* https://github.com/openshift/api/pull/1012
* https://github.com/openshift/apiserver-library-go/pull/63
* https://github.com/openshift/client-go/pull/196

damemi commented:
/retest

damemi commented:
/retest

damemi commented:
/retest

damemi commented:
/retest

soltysh commented:
it's ok to squash these together:

soltysh commented:
Feel free to squash all of commits touching `openshift-hack/` dir into a single commit.

soltysh commented:
Don't forget to update `openshift-hack/images/hyperkube/Dockerfile.rhel` with 1.19.14

damemi commented:
Updated with feedback:
* squashed openshift-related changes to https://github.com/openshift/kubernetes/pull/962/commits/19b195439cc6f0b67e79a3f1caaaa2d2529922fc
* Squashed multiple commits for the same PR
* Updated dockerfile.rhel version

Once my other bumps are cherry-pick approved, I can update the vendor here to use them

soltysh commented:
Last 2 bits:
1. you're missing this commit `c085042440e			UPSTREAM: <drop>: enable TCP_USER_TIMEOUT on client connections	https://github.com/openshift/kubernetes/commit/c085042440e?w=1` the drop is for the next release and not 4.6, double check that with @p0lyn0mial who originally authored it
2. you're missing `hack/update-vendor.sh` to pass verify

soltysh commented:
/override ci/prow/verify-commits
this will never pass on k8s bump

soltysh reviewed (approved):
/approve
/lgtm
/hold
for api team final approval

p0lyn0mial commented:
> you're missing this commit c085042440e UPSTREAM: <drop>: enable TCP_USER_TIMEOUT on client connections https://github.com/openshift/kubernetes/commit/c085042440e?w=1 the drop is for the next release and not 4.6, double check that with @p0lyn0mial who originally authored it

please bring back that commit, the `http.ping` was added in 1.20+, which means we can drop the commit in 4.7

ingvagabund commented:
@p0lyn0mial checking https://github.com/openshift/kubernetes/pull/962/commits `UPSTREAM: : enable TCP_USER_TIMEOUT on client connections` is there at the very bottom.

sttts commented:
Have verified those commits with 1 and 2 in the spreadsheet. Lgtm.

soltysh commented:
> please bring back that commit, the `http.ping` was added in 1.20+, which means we can drop the commit in 4.7

As Jan pointed out, it's there, I've asked Mike yesterday to bring it back. `<drop>` commits although misleading are tricky, and since this is only within patch version we need to keep it.
